### DISCUSSION TITLE
How to prevent user from setting roles he does not already have

### DISCUSSION THREAD

**Discussion by alonsoac** [LiveHelperChat/livehelperchat#1883] [General]
I am trying to give a user permission to create and edit other users but this allows him to change his roles and give himself or other users more permissions.  It would be necessary to have a permission to allow user to assign all roles or only ones he already has, or add a limitation with list of roles he can assign.

**Comment by remdex** [comment]
You can't do that way, but you can limit with what groups he can work.

But you can define `Teamleader group` and set with what groups it can work
![image](https://user-images.githubusercontent.com/1146085/211750319-8b26ff8e-f8a7-4bb1-9aee-002091f9768c.png)

This or next week I'll try to prepare the article.

**Comment by remdex** [comment]
But in short there is no way to setup permission that he can adjust permissions themself, you can setup that he can setup only specific departments or assign specific user groups, but not edit roles or their functions.

**Comment by remdex** [comment]
Please read this new article
https://doc.livehelperchat.com/docs/users/advanced-permissions
